core/src/main/java/org/infinispan/commands/functional/ReadWriteKeyCommand.java

```diff
@@ -130,4 +128,26 @@ public void init(ComponentRegistry componentRegistry) {
          ((InjectableComponent) f).inject(componentRegistry);
       }
    }
+
+   @Override
+   public final boolean isReturnValueExpected() {
+      return true;
+   }
+
+   @Override
+   public Metadata getMetadata() {
```

Comment:
Christmas? You're very optimistic :)
No kidding; there are two issues:
1) UnsupportedOperationException with SC: I think that it's okay to have `merge` failing in SC for a while. That should not block this PR, IMO.
2) Adding `MetaAwareX`: While I've noticed that interceptors *might* want to add some extra Metadata when transfering functional command from node to node, I still firmly believe that "functional commands should not be aware of carrying any *data*, just the function". The way to perform this is shown in #5232 - if you don't want to wait until this PR gets merged, you can cherry-pick the changes in `MetaParams` separately (reviewers already seem to be content with that part).